src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplateCallbackTests.java

```diff
@@ -15,258 +15,86 @@
-import static org.assertj.core.api.Assertions.*;
 import static org.mockito.Mockito.*;
 
-import lombok.AllArgsConstructor;
-import lombok.Data;
-import lombok.NoArgsConstructor;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Iterator;
-import java.util.List;
-
 import org.elasticsearch.action.bulk.BulkItemResponse;
 import org.elasticsearch.action.bulk.BulkRequest;
 import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.get.MultiGetItemResponse;
+import org.elasticsearch.action.get.MultiGetRequest;
+import org.elasticsearch.action.get.MultiGetResponse;
 import org.elasticsearch.action.index.IndexRequest;
 import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.search.MultiSearchRequest;
+import org.elasticsearch.action.search.MultiSearchResponse;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchScrollRequest;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.bytes.BytesArray;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.mockito.junit.jupiter.MockitoSettings;
 import org.mockito.quality.Strictness;
-import org.springframework.data.annotation.Id;
-import org.springframework.data.elasticsearch.core.event.AfterSaveCallback;
-import org.springframework.data.elasticsearch.core.mapping.IndexCoordinates;
-import org.springframework.data.elasticsearch.core.query.BulkOptions;
-import org.springframework.data.elasticsearch.core.query.IndexQuery;
-import org.springframework.data.mapping.callback.EntityCallbacks;
-import org.springframework.lang.Nullable;
-import org.springframework.util.CollectionUtils;
+
+import com.google.common.collect.ImmutableMap;
 
 /**
  * @author Roman Puchkovskiy
  */
 @ExtendWith(MockitoExtension.class)
 @MockitoSettings(strictness = Strictness.LENIENT)
-public class ElasticsearchRestTemplateCallbackTests {
-
-	private ElasticsearchRestTemplate template;
+class ElasticsearchRestTemplateCallbackTests extends AbstractElasticsearchTemplateCallbackTests {
 
 	@Mock private RestHighLevelClient client;
 
 	@Mock private IndexResponse indexResponse;
 	@Mock private BulkResponse bulkResponse;
 	@Mock private BulkItemResponse bulkItemResponse;
+	@Mock private GetResponse getResponse;
+	@Mock private MultiGetResponse multiGetResponse;
+	@Mock private MultiGetItemResponse multiGetItemResponse;
+	@Mock private MultiSearchResponse.Item multiSearchResponseItem;
 
+	@SuppressWarnings("deprecation") // we know what we test
 	@BeforeEach
 	public void setUp() throws Exception {
 		template = new ElasticsearchRestTemplate(client);
+		initTemplate(template);
```

Comment:
```suggestion
		initTemplate(new ElasticsearchRestTemplate(client));
```

with the change in the base class

Reply:
Fixed